From b82c7b4beb7bba67508b4b8f6ba701c5781357b4 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Mon, 10 Sep 2018 17:17:01 +0300
Subject: [PATCH 296/706] enetc: don't dump stack on command error from HW

print just an error message instead.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit d05e9de912e682adea24d6a6a07736c8a9ce1304)
(cherry picked from commit 9dd52ac3015beff3b1c751de8a39e3910c53ed83)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../net/ethernet/freescale/enetc/enetc_cbdr.c | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_cbdr.c b/drivers/net/ethernet/freescale/enetc/enetc_cbdr.c
index e1bf3a422ae6..5dfb7c26a2d3 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_cbdr.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_cbdr.c
@@ -34,18 +34,17 @@
 
 #include "enetc.h"
 
-static void enetc_clean_cbdr(struct enetc_si *si)
+static int enetc_clean_cbdr(struct enetc_si *si)
 {
 	struct enetc_cbdr *ring = &si->cbd_ring;
 	struct enetc_cbd *dest_cbd;
-	int i;
+	int i, err = 0;
 
 	i = ring->next_to_clean;
 
 	while (enetc_rd_reg(ring->cisr) != i) {
 		dest_cbd = ENETC_CBD(*ring, i);
-		if (dest_cbd->status_flags & ENETC_CBD_STATUS_MASK)
-			WARN_ON(1);
+		err = dest_cbd->status_flags & ENETC_CBD_STATUS_MASK;
 
 		memset(dest_cbd, 0, sizeof(*dest_cbd));
 
@@ -53,6 +52,7 @@ static void enetc_clean_cbdr(struct enetc_si *si)
 	}
 
 	ring->next_to_clean = i;
+	return err;
 }
 
 static int enetc_send_cmd(struct enetc_si *si, struct enetc_cbd *cbd,
@@ -60,7 +60,7 @@ static int enetc_send_cmd(struct enetc_si *si, struct enetc_cbd *cbd,
 {
 	struct enetc_cbdr *ring = &si->cbd_ring;
 	struct enetc_cbd *dest_cbd;
-	int i;
+	int i, err;
 
 	if (!ring->bd_base)
 		return -EIO;
@@ -98,8 +98,13 @@ static int enetc_send_cmd(struct enetc_si *si, struct enetc_cbd *cbd,
 			return ENETC_CMD_TIMEOUT;
 	}
 
-	if (!async)
-		enetc_clean_cbdr(si);
+	if (!async) {
+		err = enetc_clean_cbdr(si);
+		if (err) {
+			dev_warn(&si->pdev->dev, "CMD err %04x for cmd %04x\n",
+				err, cbd->cmd);
+		}
+	}
 
 	return ENETC_CMD_OK;
 }
-- 
2.17.1

